tools/swift-plugin-server/CMakeLists.txt

```diff
@@ -23,8 +32,32 @@ if (SWIFT_BUILD_SWIFT_SYNTAX)
       SwiftSyntaxMacroExpansion
       SwiftCompilerPluginMessageHandling
       swiftLLVMJSON
+      SwiftPluginServerSupport
   )
   target_include_directories(swift-plugin-server PRIVATE
     Sources/CSwiftPluginServer/include
   )
+
+  add_pure_swift_host_tool(swift-wasm-plugin-server
+    Sources/swift-wasm-plugin-server/swift-wasm-plugin-server.swift
+    Sources/swift-wasm-plugin-server/JSCWasmPlugin.swift
+    DEPENDENCIES
+      $<TARGET_OBJECTS:_swiftCSwiftPluginServer>
+    SWIFT_COMPONENT
+      compiler
+    SWIFT_DEPENDENCIES
+      SwiftCompilerPluginMessageHandling
+      swiftLLVMJSON
+      SwiftPluginServerSupport
+  )
+  target_include_directories(swift-wasm-plugin-server PRIVATE
+    Sources/CSwiftPluginServer/include
+  )
+  if(SWIFT_HOST_VARIANT_SDK IN_LIST SWIFT_DARWIN_PLATFORMS)
+    add_custom_command(TARGET swift-wasm-plugin-server POST_BUILD COMMAND
+      codesign -fs - $<TARGET_FILE:swift-wasm-plugin-server> --entitlements ${CMAKE_CURRENT_SOURCE_DIR}/allow-jit.plist
```

Comment:
not sure if this is sufficient: do we need a corresponding codesign invocation in build_script as well?